presto-main/src/test/java/com/facebook/presto/operator/scalar/TestHyperLogLogFunctions.java

```diff
@@ -0,0 +1,183 @@
+package com.facebook.presto.operator.scalar;
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.facebook.presto.Session;
+import com.facebook.presto.SessionTestUtils;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.testing.MaterializedResult;
+import com.google.common.io.BaseEncoding;
+import io.airlift.slice.Slice;
+import io.airlift.stats.cardinality.HyperLogLog;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+public class TestHyperLogLogFunctions
+        extends AbstractTestFunctions
+{
+    private TestHyperLogLogFunctions() {}
+
+    private static final int NUMBER_OF_BUCKETS = 32768;
+
+    @Test
+    public void testApproxNullArray()
+    {
+        String projection = "cardinality(merge_hll(NULL))";
+
+        Session session = SessionTestUtils.TEST_SESSION;
+        FeaturesConfig featuresConfig = new FeaturesConfig();
+        LocalQueryRunner runner = new LocalQueryRunner(session, featuresConfig);
+
+        MaterializedResult result = runner.execute("SELECT " + projection);
+
+        assertNull(result.getOnlyValue());
+    }
+
+    @Test
+    public void testApproxMultipleNullColumns()
+    {
+        String projection = "cardinality(merge_hll(ARRAY[null, null, null]))";
```

Comment:
Can you use the `assertFunction()` method to do the test? It basically wraps the logic so that you don't need to create the query runner manually. You can check other test cases as examples, e.g. `TestArrayFilterFunction`.

Reply:
I wrote a new private function (very similar to assertFunction()) because I needed the specific cardinality value returned by the Presto engine in order to create my range of acceptable values (+- delta). assertFunction() only returns a boolean which states if the function equals a specific output, but not a range (as needed here due to the error associated with HyperLogLogs).